providers/default/macs/siphash_prov.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/opensslconf.h>
+#ifndef OPENSSL_NO_SIPHASH
+
+# include <string.h>
+# include <openssl/core_numbers.h>
+# include <openssl/core_names.h>
+# include <openssl/params.h>
+# include <openssl/evp.h>
+# include <openssl/err.h>
+
+# include "internal/siphash.h"
+# include "../../../crypto/siphash/siphash_local.h"
```

Comment:
Again - why is this necessary?

Reply:
Because line 43:

``` C
    SIPHASH siphash;             /* Siphash data */
```

Reply:
Which is why i moved some headers and TODO'd them when doing the digest work.

Reply:
I do remember the dispute we had ;-)

Reply:
TODO added